state/execution_test.go

```diff
@@ -0,0 +1,329 @@
+package state
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	abci "github.com/tendermint/abci/types"
+	"github.com/tendermint/basecoin/types"
+	"github.com/tendermint/go-crypto"
+)
+
+//States and Stores for tests
+var (
+	store                                 types.KVStore
+	state                                 *State
+	accsFoo, accsBar, accsFooBar, accsDup []types.PrivAccount
+	chainID                               string = "test_chain_id"
+)
+
+func makeAccs(secrets []string) (accs []types.PrivAccount) {
+
+	for _, secret := range secrets {
+		privAcc := types.PrivAccountFromSecret(secret)
+		privAcc.Account.Balance = types.Coins{{"mycoin", 1000}}
+		accs = append(accs, privAcc)
+	}
+	return accs
+}
+
+func acc2State(accs []types.PrivAccount) {
+	for _, acc := range accs {
+		state.SetAccount(acc.Account.PubKey.Address(), &acc.Account)
+	}
+}
+
+//each tx input signs the tx bytes
+func signSend(tx *types.SendTx, accs []types.PrivAccount) {
+	signBytes := tx.SignBytes(chainID)
+	for i, _ := range tx.Inputs {
+		tx.Inputs[i].Signature = crypto.SignatureS{accs[i].Sign(signBytes)}
+	}
+}
+
+//turn a list of accounts into basic list of transaction inputs
+func accs2TxInputs(accs []types.PrivAccount) []types.TxInput {
```

Comment:
Okay, created common functions between `execution_test` and `app_test` added to `types/test_helpers.go`... also cleaned up the `execution_test` a bunch while I was at it, similar test designs as `app_test` now, no global variables